### ISSUE TITLE
Add troubleshooting guide for remote clusters provisioning

### ISSUE THREAD

**Issue by eromanova** [k0rdent/docs#788]
It would be useful to have a troubleshooting guide for remote clusters covering common issues users may encounter:
1. API server inaccessibility due to incorrect k0smotron service type or missing CCM
2. Missing storage backend and the resulting errors
3. Other known issues and their symptoms
